### PR TITLE
feat: add pipelines overview docs

### PR DESCRIPTION
**Pull request by alex131290**

### CODE REVIEW

File: docs/guides/DevOps/pipelines.mdx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Overview
+
+We store all our helm charts of all the teams under the [helm-charts repository](https://github.com/MapColonies/helm-charts)
+
+This repository is a mono repo, where each team has it’s own `scope` for example, `3D` team has it’s own directory where the team is storing their helm charts.
```
Comment (netanelC) [inline docs/guides/DevOps/pipelines.mdx]:
```suggestion
This repository is a mono repo, where each team has it’s own `scope`. for example, `3D` team has it’s own directory where the team is storing their helm charts.  
```

**Review by netanelC — CHANGES_REQUESTED** [review]
Good Job!
1. It seems that this article is more of a knowledge-base article than a guide article. Consider moving it.
2. Separate the pipelines to a different pages.
3. Add also a short live with the triggered pipelines
4. You have some typos, I recommend installing the "Code Spell Checker" extension in VS Code
5. Extend the git step in unbundler pipeline. Add also a "before and after" in this pipeline for better visualization and understanding

File: docs/guides/DevOps/img/ci-cd-workflows.png
Comment (netanelC) [inline docs/guides/DevOps/img/ci-cd-workflows.png:1]: You are not using this file

**Review by netanelC — APPROVED** [review]

File: docs/knowledge-base/DevOps/pipelines.mdx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+The workflows are being triggered automatically.  
+If we take a look at a sample repo `store-trigger` it has a workflow called `Build and push artifacts` , where the source of this can be found [here](https://github.com/MapColonies/store-trigger/blob/master/.github/workflows/build-and-push.yaml)
+
+The workflow is building the dokcer image of the service, tagging it and pushing it to `Azure ACR`, along with the docker image it's also pushing the helm package which is part of the repository.  
+Afterwards it's updating the `artifacts.json` which is associated with the team's `scope` under the [helm-charts repository](https://github.com/MapColonies/helm-charts) with the new releases which have been just released.  
+
+### CD part
+
+We're utilizing [Jenkins](https://jenkins.mapcolonies.net/) for this purpose.
```
Comment (michalby24) [inline docs/knowledge-base/DevOps/pipelines.mdx]: We use the Jenkins in all our pipelines, I suggest moving it to the top of the doc

File: docs/knowledge-base/DevOps/pipelines.mdx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+* If you wish to skip a deployment you'll need to have `"skip": true` in the chart you wish to skip.
+* When the pipeline will start it'll send a slack message to the `<team>-notifications` channel based on chart location(s) which are going to be deployed.  
+![jenkins auto deploy slack start message](./img/jenkins-auto-deploy-2.png) 
+* After the pipeline will finish it'll send another message to the `<team>-notifications` channel based on chart location(s) which have been deployed with the status of the deployment.  
+
+## Bundler/Unbundler
+
+The bundler and unbundler workflows are intended to help our dev team to bring in the artifacts which were built in the release flow in an easy way to our internal deployments.
+
+We utilize Jenkins for the bundler and unbundler.
```
Comment (michalby24) [inline docs/knowledge-base/DevOps/pipelines.mdx]: We use the Jenkins in all our pipelines, I suggest moving it to the top of the doc

File: docs/knowledge-base/DevOps/pipelines.mdx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+We can see the diagram of the Release workflow below
+
+![release workflow diagram](./img/release-workflow-diagram.png)
+
+### CI part
+
+The release workflow begins with a new release or tag push on each service repository and it's happening via Github Actions workflows.
+
+The workflows are being triggered automatically.  
+If we take a look at a sample repo `store-trigger` it has a workflow called `Build and push artifacts` , where the source of this can be found [here](https://github.com/MapColonies/store-trigger/blob/master/.github/workflows/build-and-push.yaml)
```
Comment (michalby24) [inline docs/knowledge-base/DevOps/pipelines.mdx]: Here’s a clearer version of that sentence:
You provided a link to a workflow, but it references two other workflows where the actual tasks are being performed. You should provide links to those workflows, as they contain the relevant code.

File: docs/knowledge-base/DevOps/pipelines.mdx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+The workflow is building the dokcer image of the service, tagging it and pushing it to `Azure ACR`, along with the docker image it's also pushing the helm package which is part of the repository.  
+Afterwards it's updating the `artifacts.json` which is associated with the team's `scope` under the [helm-charts repository](https://github.com/MapColonies/helm-charts) with the new releases which have been just released.  
+
+### CD part
+
+We're utilizing [Jenkins](https://jenkins.mapcolonies.net/) for this purpose.
+
+#### How to login?
```
Comment (michalby24) [inline docs/knowledge-base/DevOps/pipelines.mdx]: This is relevant to Jenkins in general and should be placed at the top in a preview section.

File: docs/knowledge-base/DevOps/pipelines.mdx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  },
+  "git": [
+    "MapColonies/helm-charts"
+  ]
+}
+```  
+
+The bundler pipeline will bundle all the artifacts (docker tar balls, helm packages, git repos) into 1 zip file.
+
+Once the bundler is done it'll send a slack message to the `<team>-notifications` channel based on the user logged in.
```
Comment (michalby24) [inline docs/knowledge-base/DevOps/pipelines.mdx]: I would have also included where the blob is stored and how to locate it.

Comment (alex131290) [reply docs/knowledge-base/DevOps/pipelines.mdx]: @michalby24 i think the end user shouldn't be aware as it's not relevant to them as they are getting a pre-signed url

File: docs/knowledge-base/DevOps/pipelines.mdx (hunk 2 above)
Comment (alex131290) [reply docs/knowledge-base/DevOps/pipelines.mdx]: @michalby24 we use jenkins in some of the pipelines and some of them are implemented via GHA, so the moving it to another place wouldn't make sense.

File: docs/knowledge-base/DevOps/pipelines.mdx [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+![Bundler slack message](./img/bundler-slack.png)
+
+
+### Unbundler pipeline
+
+The unbundler pipeline is happening internally in every network we have.
```
Comment (CptSchnitz) [inline docs/knowledge-base/DevOps/pipelines.mdx]: dont write internally, use somthing like other networks

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
First of all good job! it's not easy to write docs.
Because the topic is so big and important, I think it should be separated to multiple files so each one is more focused and easier to read.
I suggest the following files
- jenkins
- code repos flows (push to registry and such)
- CD flow
- bundler
- Maybe a landing page with links/overview
wdyt @netanelC?
after the split it will be easier to read and review each part.

File: docs/knowledge-base/DevOps/pipelines.mdx (hunk 7 above)
Comment (alex131290) [reply docs/knowledge-base/DevOps/pipelines.mdx]: @CptSchnitz fixed

File: docs/knowledge-base/DevOps/pipelines/ci.mdx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+toc_max_heading_level: 6
+
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Overview
+
+The CI pipelines are responsible for validating, compiling and building the artifacts and publishing them to the various registries.
```
Comment (CptSchnitz) [inline docs/knowledge-base/DevOps/pipelines/ci.mdx]: Add continues integration

File: docs/knowledge-base/DevOps/pipelines/ci.mdx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    - Umbrella chart
+    - deployment.json
+    - artifacts.json
+```
+
+## Release Workflow
+
+![release workflow diagram](./img/release-workflow-diagram.png)
+
+The release workflow begins with a new release or tag push on each service repository and it's happening via Github Actions workflows.
```
Comment (CptSchnitz) [inline docs/knowledge-base/DevOps/pipelines/ci.mdx]:
```suggestion
The release workflow begins with a new release or tag push on each service repository and is executed using Github Actions workflows.
```

File: docs/knowledge-base/DevOps/pipelines/ci.mdx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+The workflows are being triggered automatically.  
+If we take a look at a sample repo `store-trigger` it has a workflow called `Build and push artifacts` , where the source of this can be found [here](https://github.com/MapColonies/store-trigger/blob/master/.github/workflows/build-and-push.yaml)
+
+The workflow is building the dokcer image of the service, tagging it and pushing it to `Azure ACR`, along with the docker image it's also pushing the helm package which is part of the repository.
```
Comment (CptSchnitz) [inline docs/knowledge-base/DevOps/pipelines/ci.mdx]:
```suggestion
The workflow is building the docker image of the service, tagging it and pushing it to `Azure ACR`. Along with the docker image the flow also builds and pushes the helm package which is part of the repository.  
```

File: docs/knowledge-base/DevOps/pipelines/cd.mdx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+We have 2 options for deployment:
+
+1. Manual deployment
+2. Auto Deploy
+
+### Auto deployment
+
+The auto deployment process is currently monitoring the following branches: `master` and `qa`. On each commit to the [helm-charts repository](https://github.com/MapColonies/helm-charts) on the monitored branch the pipeline is kicking off.
+
+The pipeline can be found [here](https://jenkins.mapcolonies.net/job/Deploy%20Chart%20Polling%20-%20Multibranch/)
```
Comment (CptSchnitz) [inline docs/knowledge-base/DevOps/pipelines/cd.mdx]:
```suggestion
The pipeline can be found [here](https://jenkins.mapcolonies.net/job/Deploy%20Chart%20Polling%20-%20Multibranch/).
```

File: docs/knowledge-base/DevOps/pipelines/cd.mdx [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+The pipeline can be found [here](https://jenkins.mapcolonies.net/job/Deploy%20Chart%20Polling%20-%20Multibranch/)  
+As you can see in the overview page of the pipeline there are 2 branches which it's monitoring.
+
+![jenkins auto deploy overview](./img/jenkins-auto-deploy-1.png)  
+
+#### How the auto deploy pipeline is working?
+
+* The pipeline is first of all looking at changes introduced in each commit with a `git diff`.
+* After finding the `diff` it'll look at the `deployment.json` in the `scope` (each team's directory in the `helm-charts` repo) directories.
+* Here's an example of `deployment.json` file
```
Comment (CptSchnitz) [inline docs/knowledge-base/DevOps/pipelines/cd.mdx]:
```suggestion
* Here's an example of `deployment.json` file.
```

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**